logstash-core/src/main/java/org/logstash/execution/PipelineReporterExt.java

```diff
@@ -267,6 +287,11 @@ public IRubyObject methodMissing(final ThreadContext context, final IRubyObject
             return data.op_aref(context, method);
         }
 
+        @JRubyMethod(name = "respond_to_missing?")
```

Comment:
TIL: `RubyObjects` interface has a `respond_to_missing?` and it seems to me it is because we used `batch.respondsTo("size")`, right?

Reply:
This reporter snapshot defines a `method_missing`, so [it _should_ define a `respond_to_missing?`](https://thoughtbot.com/blog/always-define-respond-to-missing-when-overriding).

The new specs attempt to validate that the snapshot has attributes, which fails unless we properly indicate that it is capable of responding to `inflight_count`.